service/resource/configmap/update.go

```diff
@@ -0,0 +1,13 @@
+package configmap
+
+import (
+	"context"
+)
+
+func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
+	return nil, nil, nil, nil
+}
+
+func (r *Resource) ProcessUpdateState(ctx context.Context, obj, updateState interface{}) error {
+	return nil
+}
```

Comment:
I think I like this separation in dedicated files.

Reply:
I just found having all the different states in one file required a lot of mental hopping around. With separate files, I can focus a bit better.